src/main/java/net/javadiscord/javabot/Bot.java

```diff
@@ -59,33 +68,62 @@
  * The main class where the bot is initialized.
  */
 @Slf4j
+@SpringBootApplication
+@ComponentScan(includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {SlashCommand.class, ContextCommand.class}))
 public class Bot {
 
+	@Getter
 	private static BotConfig config;
```

Comment:
You might want to make these non-static and initialize them and [instantiating them in `@Configuration` classes](https://docs.spring.io/spring-javaconfig/docs/1.0.0.M4/reference/html/ch02s02.html).

That would allow autowiring of any of those fields.

Reply:
Does this even work with the bots' config? The Spring App is initialized at the end of the main method. However, JDA and stuff need the BotConfig before that. Can I just move the Spring App init thingy to the top?

Reply:
You could move most of the stuff from the main method to a `@PostConstruct` method of a spring managed bean (for example the `Bot` class).

Reply:
And how/were do I initialize all those Beans? (Like the JDA one)